lib/Controller/LoginController.php

```diff
@@ -467,10 +467,13 @@ public function code(string $state = '', string $code = '', string $scope = '',
 
 		$autoProvisionAllowed = (!isset($oidcSystemConfig['auto_provision']) || $oidcSystemConfig['auto_provision']);
 
-		// in case user is provisioned by user_ldap, userManager->search() triggers an ldap search which syncs the results
-		// so new users will be directly available even if they were not synced before this login attempt
-		$this->userManager->search($userId);
-		$this->ldapService->syncUser($userId);
+		if (!$this->provisioningService->hasOidcUserProvisitioned($userId) && $this->ldapService->isLDAPEnabled()) {
```

Comment:
Not sure this condition is enough. What if the user exists in the user_oidc backend but auto-provisioning is disabled and we still expect to find the user via LDAP. We need to force-sync then too. Maybe this exact case never happens but maybe we are missing some other cases.

IMO We have to make sure to trigger those when (auto-provisioning is disabled) OR (soft-autoprovisioning is enabled AND the user is not found in the user_oidc backend).
In other words: In all the cases where we might need to find the user in another backend.

Reply:
All right, then lets add both cases, is still fine with me for the case I'm looking at where ldap is not used